File: Projects/Offensive/Wi-Fi security tester/wifi_tester.c
/* wifi_tester.c — Educational Wi-Fi Security Analyzer (safe)
   Compile: gcc -o wifi_tester wifi_tester.c
   Run: sudo ./wifi_tester
*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

#define BUF_SIZE 8192

void analyze_line(const char *line) {
    if (strstr(line, "Encryption key:off")) {
        printf("⚠️  Open network detected!\n");
    }
    if (strstr(line, "WEP")) {
        printf("⚠️  Weak WEP encryption found!\n");
    }
}

int main() {
    FILE *fp;
    char buf[BUF_SIZE];

    printf("Scanning Wi-Fi networks...\n");
    fp = popen("iwlist scan 2>/dev/null", "r");
    if (!fp) {
        perror("popen");
        return 1;
    }

    while (fgets(buf, sizeof(buf), fp)) {
        if (strstr(buf, "Cell ")) printf("\n--- Network ---\n");
        if (strstr(buf, "ESSID:")) {
            char *p = strchr(buf, '"');
            if (p) printf("SSID: %s\n", p+1);
        }
        if (strstr(buf, "Encryption key:") || strstr(buf, "IE: IEEE 802.11i")) {
            analyze_line(buf);
        }
        if (strstr(buf, "Quality=")) {
            printf("%s", buf);
        }
    }

    pclose(fp);
    printf("\nDone.\n");
    return 0;
}

File: Projects/Offensive/Wi-Fi security tester/wifi_tester.py
#!/usr/bin/env python3
"""
wifi_tester.py — Educational Wi-Fi Security Analyzer
Safe for home/lab use.

Functions:
- Scan nearby networks using nmcli/iwlist
- Analyze encryption type, SSID patterns, and signal quality
- Flag weak security (Open/WEP)
- Optional dictionary strength test (offline)
"""

import os, re, json, subprocess

def scan_wifi():
    """Use nmcli to scan Wi-Fi networks (Linux)"""
    try:
        result = subprocess.check_output(
            ["nmcli", "-t", "-f", "SSID,SECURITY,SIGNAL", "device", "wifi", "list"],
            text=True
        )
    except Exception as e:
        print("Scan failed:", e)
        return []
    networks = []
    for line in result.strip().split("\n"):
        if not line.strip(): continue
        parts = line.split(":")
        if len(parts) >= 3:
            ssid, sec, signal = parts[0], parts[1] or "OPEN", parts[2]
            networks.append({
                "ssid": ssid or "<hidden>",
                "security": sec,
                "signal": int(signal)
            })
    return networks

def analyze_networks(networks):
    report = []
    for n in networks:
        score, issues = 100, []
        sec = n["security"].upper()
        ssid = n["ssid"]
        if sec == "OPEN" or "WEP" in sec:
            score -= 70
            issues.append("Weak encryption (Open/WEP)")
        if re.match(r"(?i)^(linksys|tp-link|netgear|dlink|huawei|tplink)", ssid):
            score -= 15
            issues.append("Default SSID detected")
        if n["signal"] < 40:
            score -= 10
            issues.append("Weak signal (may reduce reliability)")
        report.append({
            "SSID": ssid,
            "Security": sec,
            "Signal": n["signal"],
            "Score": max(score, 0),
            "Issues": issues or ["OK"]
        })
    return report

def show_report(report):
    print("\n=== Wi-Fi Security Report ===")
    for r in report:
        print(f"\nSSID: {r['SSID']}")
        print(f"  Security: {r['Security']}")
        print(f"  Signal: {r['Signal']}%")
        print(f"  Score: {r['Score']}/100")
        for issue in r["Issues"]:
            print(f"   - {issue}")

if __name__ == "__main__":
    nets = scan_wifi()
    if not nets:
        print("No networks found or scanning failed.")
        exit()
    report = analyze_networks(nets)
    show_report(report)
    with open("wifi_report.json", "w") as f:
        json.dump(report, f, indent=2)
    print("\nReport saved to wifi_report.json")
